From: Liam Howlett <liam.howlett@oracle.com>
Subject: mm.h: fix "mm: Add temporary vma iterator versions of vma_merge(), split_vma(), and __split_vma()
Date: Mon, 9 Jan 2023 20:53:22 +0000

The extern definition of __split_vma() was removed too early.  Re-add it
and remove it once the users are all converted.

Link: https://lkml.kernel.org/r/20230109205300.955019-1-Liam.Howlett@oracle.com
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Reported-by: SeongJae Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mm.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/mm.h~mm-add-temporary-vma-iterator-versions-of-vma_merge-split_vma-and-__split_vma-fix
+++ a/include/linux/mm.h
@@ -2825,6 +2825,8 @@ extern struct vm_area_struct *vmi_vma_me
 	struct file *, pgoff_t, struct mempolicy *, struct vm_userfaultfd_ctx,
 	struct anon_vma_name *);
 extern struct anon_vma *find_mergeable_anon_vma(struct vm_area_struct *);
+extern int __split_vma(struct mm_struct *, struct vm_area_struct *,
+			       unsigned long addr, int new_below);
 extern int vmi__split_vma(struct vma_iterator *vmi, struct mm_struct *,
 	struct vm_area_struct *, unsigned long addr, int new_below);
 extern int split_vma(struct mm_struct *, struct vm_area_struct *,
_
